File: internal/yjsworker/transports/status/started_router.ts
import type { Hono } from "hono";

export function configureStartedRouter(
  app: Hono,
  isStarted: () => boolean
): void {
  app.get("/startedz", context => {
    if (!isStarted()) {
      return context.body(null, 503);
    }

    return context.body(null, 200);
  });
}

File: internal/apigateway/transports/status/started_router.go
package status

import (
	"net/http"

	gin "github.com/gin-gonic/gin"
)

func ConfigureStartedRouter(router gin.IRouter, isStarted func() bool) {
	router.GET("/startedz", func(ctx *gin.Context) {
		if !isStarted() {
			ctx.Status(http.StatusServiceUnavailable)
			return
		}

		ctx.Status(http.StatusOK)
	})
}
